Sources/ComposableArchitecture/UIKit/ForEachUIKit.swift

```diff
@@ -0,0 +1,76 @@
+extension Store where State: Equatable {
+
+  public func forEach<LocalState: Equatable, LocalAction>(
```

Comment:
These were shamelessly copied from:
https://github.com/pointfreeco/swift-composable-architecture/discussions/351#discussioncomment-312688

Decided to add them anyway since they are useful if one wants to generate all the child stores at once for every children state update.. However they are not very useful when using diffing (which is the main scope of this PR), so I am ok with leaving them to a separate PR.